libvast/vast/transform.hpp

```diff
@@ -26,15 +39,23 @@ class transform {
 
   void add_step(transform_step_ptr step);
 
-  caf::expected<table_slice> apply(table_slice&&) const;
+  // FIXME: Convinence function
+  [[nodiscard]] caf::error add_slice(table_slice&&);
+  // FIXME: Convinence function
+  // The result is not ordered by offset
+  [[nodiscard]] caf::expected<std::vector<table_slice>> finish_slice();
```

Comment:
Please provide API docs.

Why do we need the `_slice` suffix for these member functions? If the non-suffixed version is private, shouldn't the naming rather be the other way around?